apps/meteor/app/livechat/imports/server/rest/rooms.ts

```diff
@@ -1,54 +1,53 @@
-import { Match, check } from 'meteor/check';
+import { isGETLivechatRoomsParams } from '@rocket.chat/rest-typings';
 
 import { API } from '../../../../api/server';
 import { findRooms } from '../../../server/api/lib/rooms';
 import { hasPermission } from '../../../../authorization/server';
 
-const validateDateParams = (property, date) => {
+const validateDateParams = (property: string, date?: string) => {
+	let parsedDate: { start?: string; end?: string } | undefined = undefined;
 	if (date) {
-		date = JSON.parse(date);
+		parsedDate = JSON.parse(date) as { start?: string; end?: string };
 	}
-	if (date && date.start && isNaN(Date.parse(date.start))) {
+
+	if (parsedDate?.start && isNaN(Date.parse(parsedDate.start))) {
 		throw new Error(`The "${property}.start" query parameter must be a valid date.`);
 	}
-	if (date && date.end && isNaN(Date.parse(date.end))) {
+	if (parsedDate?.end && isNaN(Date.parse(parsedDate.end))) {
 		throw new Error(`The "${property}.end" query parameter must be a valid date.`);
 	}
-	return date;
+	return parsedDate;
 };
 
 API.v1.addRoute(
 	'livechat/rooms',
-	{ authRequired: true },
+	{ authRequired: true, validateParams: isGETLivechatRoomsParams },
 	{
 		async get() {
 			const { offset, count } = this.getPaginationItems();
 			const { sort, fields } = this.parseJsonQuery();
 			const { agents, departmentId, open, tags, roomName, onhold } = this.requestParams();
-			let { createdAt, customFields, closedAt } = this.requestParams();
-			check(agents, Match.Maybe([String]));
-			check(roomName, Match.Maybe(String));
-			check(departmentId, Match.Maybe(String));
-			check(open, Match.Maybe(String));
-			check(onhold, Match.Maybe(String));
-			check(tags, Match.Maybe([String]));
-			check(customFields, Match.Maybe(String));
+			const { createdAt, customFields, closedAt } = this.requestParams();
 
-			createdAt = validateDateParams('createdAt', createdAt);
-			closedAt = validateDateParams('closedAt', closedAt);
+			const createdAtParam = validateDateParams('createdAt', createdAt);
+			const closedAtParam = validateDateParams('closedAt', closedAt);
 
 			const hasAdminAccess = hasPermission(this.userId, 'view-livechat-rooms');
 			const hasAgentAccess = hasPermission(this.userId, 'view-l-room') && agents?.includes(this.userId) && agents?.length === 1;
 			if (!hasAdminAccess && !hasAgentAccess) {
 				return API.v1.unauthorized();
 			}
 
+			let parsedCf: { [key: string]: string } | undefined = undefined;
 			if (customFields) {
 				try {
-					const parsedCustomFields = JSON.parse(customFields);
-					check(parsedCustomFields, Object);
+					const parsedCustomFields = JSON.parse(customFields) as { [key: string]: string };
+					if (typeof parsedCustomFields !== 'object' && parsedCustomFields !== null && !Array.isArray(parsedCustomFields)) {
```

Comment:
Didn't get the logic over here. Shouldn't it just be 
```
if (typeof parsedCustomFields !== 'object') {
	throw new Error('Invalid Custom fields')
}
```

Reply:
That extra ifs are because of this:
![image](https://user-images.githubusercontent.com/11577696/191034962-b4bdd5ba-13b1-4781-aa29-bcc91724160c.png)

Both params will be parsed as they're valid JSON, but they'll make the endpoint fail since it's not what we're expecting.